### PR TITLE
BaSP-TG-05:Admin Endpoints (GetOne, Update, Delete)

### PR DESCRIPTION
**Pull request by LucianoFranchino**

### CODE REVIEW

File: src/controllers/admins.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import Admin from '../models/Admin';
+
+const getAdminById = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const admins = await Admin.findById(id);
+    if (!admins) {
+      return res.status(404).json({
+        message: 'Admin does not exists',
+        error: false,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:10]: i think this prop is not necessary

File: src/controllers/admins.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      });
+    }
+    return res.status(200).json({
+      message: 'Admin Found',
+      data: admins,
+      error: false,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:16]: same here

File: src/controllers/admins.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    });
+  } catch (error) {
+    return res.json({
```
Comment (ignacioaristo) [inline src/controllers/admins.js:19]: the status code is missing.

File: src/controllers/admins.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      message: 'An error ocurred',
```
Comment (ignacioaristo) [inline src/controllers/admins.js:20]: we can handle the message in a better way, for example: `message: 'Admin does not exists': ${error},`

File: src/controllers/admins.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      error,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:21]: same here, i think is not necessary

File: src/controllers/admins.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    });
+  }
+};
+
+const editAdmin = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const result = await Admin.findByIdAndUpdate(
+      { _id: id },
+      { ...req.body },
```
Comment (ignacioaristo) [inline src/controllers/admins.js:31]: spread operation is necessary here? I think `req.body` is enough

File: src/controllers/admins.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    if (!result) {
+      return res.status(404).json({
+        message: 'Admin does not exists',
+        error: false,
+      });
+    }
+    return res.status(200).json({
+      message: `Admin with ID ${id} edited.`,
+      data: result,
+      error: false,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:43]: same here.

File: src/controllers/admins.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    });
+  } catch (error) {
+    return res.json({
```
Comment (ignacioaristo) [inline src/controllers/admins.js:46]: status code is missing.

File: src/controllers/admins.js [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+      message: 'An error occurred',
```
Comment (ignacioaristo) [inline src/controllers/admins.js:47]: same error handling.

File: src/controllers/admins.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+      });
+    }
+    return res.status(200).json({
+      message: `Admin with ID ${id} deleted.`,
+      data: result,
+      error: false,
+    });
+  } catch (error) {
+    return res.json({
+      message: 'An error has occurred',
```
Comment (ignacioaristo) [inline src/controllers/admins.js:69]: same here.

File: src/controllers/admins.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+      error,
```
Comment (ignacioaristo) [inline src/controllers/admins.js:70]: same with this prop

**Review by paurinaldi — CHANGES_REQUESTED** [review]

File: src/validations/admins.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    lastName: Joi.string().min(3).required(),
+    email: Joi.string().email().required(),
+    password: Joi.string().regex(/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/).required(),
+  });
+
+  const validation = adminValidation.validate(req.body);
+  if (validation.error) {
+    return res.status(400).json({
+      message: `There was an error:${validation.error.details[0].message}`,
+      data: undefined,
```
Comment (ignacioaristo) [inline src/validations/admins.js:15]: i think is not necessary to show the data as undefined, remove it.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js (hunk 1 above)
Comment (LucianoFranchino) [reply src/controllers/admins.js:10]: All the if? or just the return function?

File: src/controllers/admins.js (hunk 6 above)
Comment (LucianoFranchino) [reply src/controllers/admins.js:31]: It seemed to me that it was more optimal that way, in the same way I can take it out

File: src/controllers/admins.js (hunk 1 above)
Comment (ignacioaristo) [reply src/controllers/admins.js:10]: Line 10

File: src/controllers/admins.js (hunk 2 above)
Comment (ignacioaristo) [reply src/controllers/admins.js:16]: Line 16.

File: src/controllers/admins.js [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+    }
+    return res.status(200).json({
+      message: `Admin with ID ${id} deleted.`,
+      data: result,
+      error: false,
+    });
+  } catch (error) {
+    return res.json({
+      message: 'Admin does not exists',
+      $: { error },
```
Comment (ignacioaristo) [inline src/controllers/admins.js:66]: Same here.

File: src/controllers/admins.js [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+        message: 'Admin does not exists',
+      });
+    }
+    return res.status(200).json({
+      message: 'Admin Found',
+      data: admins,
+    });
+  } catch (error) {
+    return res.status(400).json({
+      message: 'Admin does not exists',
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/controllers/admins.js:18]: Resolved

File: src/controllers/admins.js [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+      });
+    }
+    return res.status(200).json({
+      message: `Admin with ID ${id} edited.`,
+      data: result,
+    });
+  } catch (error) {
+    return res.status(400).json({
+      message: 'Admin does not exists',
+      $: { error },
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/controllers/admins.js:44]: Resolved

File: src/controllers/admins.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+  }
+};
+const deleteAdmin = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const result = await Admin.findByIdAndDelete(id);
+    if (!result) {
+      return res.status(404).json({
+        message: 'Admin does not exists',
+        error: false,
```
[the remark this answers is not in the document]
Comment (LucianoFranchino) [reply src/controllers/admins.js:55]: Resolved

File: src/controllers/admins.js (hunk 13 above)
Comment (LucianoFranchino) [reply src/controllers/admins.js:66]: Resolved

**Pull request merged**